modules/xmpp/ChatRoom.js

```diff
@@ -691,21 +692,28 @@ ChatRoom.prototype.removeListener = function (type, listener) {
     this.eventEmitter.removeListener(type, listener);
 };
 
-ChatRoom.prototype.remoteStreamAdded = function(data, sid, thessrc) {
-    if(this.lastPresences[data.peerjid])
-    {
-        var pres = this.lastPresences[data.peerjid];
-        var audiomuted = filterNodeFromPresenceJSON(pres, "audiomuted");
-        var videomuted = filterNodeFromPresenceJSON(pres, "videomuted");
-        data.videomuted = ((videomuted.length > 0
-            && videomuted[0]
-            && videomuted[0]["value"] === "true")? true : false);
-        data.audiomuted = ((audiomuted.length > 0
-            && audiomuted[0]
-            && audiomuted[0]["value"] === "true")? true : false);
-    }
-
-    this.eventEmitter.emit(XMPPEvents.REMOTE_TRACK_ADDED, data, sid, thessrc);
+ChatRoom.prototype.remoteTrackAdded = function(data) {
+    // Will figure out current muted status by looking up owner's presence
+    var pres = this.lastPresences[data.owner];
+    var mediaType = data.mediaType;
+    if(pres) {
+        var mutedNode = null;
+        if (mediaType === MediaType.AUDIO) {
+            mutedNode = filterNodeFromPresenceJSON(pres, "audiomuted");
+        } else if (mediaType === MediaType.VIDEO) {
+            mutedNode = filterNodeFromPresenceJSON(pres, "videomuted");
+        } else {
+            logger.warn("Unsupported media type: " + mediaType);
+            data.muted= null;
+        }
+
+        if (mutedNode) {
+            data.muted = !!(mutedNode.length > 0 &&
```

Comment:
Why do we need to "!!" here? The expression in the parens already evaluates to a boolean.